guardian/amigo
# Upload list of installed packages to S3 at the end of a bake

## What does this change?
This adds a new step to *all* bakes on amigo which uploads a list of installed packages to a bucket specified in configuration. The uploaded file is tagged with two tags, one with the recipe ID, the other with the bake ID. Using these two parameters we can associate the package list with AMIs in amiable. This can help us answer questions such as 'what software is installed on discussion API instances' and 'are we using X library anywhere?' (harder, would require full download of the bucket contents).

## How to test
Run a bake with amigo (this is on CODE at the moment) - you should see a final step where the package list is uploaded to S3, and see the resulting file in S3. 

## How can we measure success?
With a tape measure. The longer the package list the better. Anything over 130cm counts as victory.

## Have we considered potential risks?
This change introduces a new thing to keep track of and clean up. Strictly speaking there should be a cleanup job just like we have for AMIs to delete these package lists once the associated AMI has been deleted. I'm lazy though, and these text files are pretty small.
